lang/elm/Makefile

```diff
@@ -18,85 +18,106 @@ USE_GITHUB=	yes
 		cereal-0.5.8.1 \
 		clock-0.8 \
 		colour-2.3.5 \
-		connection-0.3.0_1 \
+		connection-0.3.1_1 \
 		containers-0.5.11.0_3 \
-		cookie-0.4.4 \
+		cookie-0.4.5 \
 		cryptonite-0.26 \
 		data-default-class-0.1.2.0 \
 		digest-0.0.1.2 \
 		edit-distance-0.2.2.1_1 \
-		exceptions-0.10.2 \
-		file-embed-0.0.11 \
-		hashable-1.2.7.0_1 \
-		haskeline-0.7.5.0 \
+		exceptions-0.10.4_1 \
+		file-embed-0.0.11.2 \
+		filelock-0.1.1.4 \
+		hashable-1.3.0.0_1 \
+		haskeline-0.8.0.0 \
 		hourglass-0.2.12 \
-		http-client-0.5.14_1 \
+		HTTP-4000.3.14_1 \
+		http-client-0.6.4.1 \
 		http-client-tls-0.3.5.3_1 \
 		http-types-0.12.3 \
-		integer-logarithms-1.0.3 \
-		io-streams-1.5.1.0 \
-		io-streams-haproxy-1.0.1.0 \
+		HUnit-1.6.0.0 \
+		integer-logarithms-1.0.3_2 \
+		io-streams-1.5.1.0_1 \
+		io-streams-haproxy-1.0.1.0_2 \
 		language-glsl-0.3.0_1 \
 		lifted-base-0.2.3.12 \
-		logict-0.7.0.1 \
-		memory-0.14.18_1 \
+		memory-0.15.0_1 \
 		mime-types-0.1.0.9 \
 		monad-control-1.0.2.3 \
 		network-2.6.3.6 \
-		network-uri-2.6.1.0_1 \
+		network-uri-2.6.3.0 \
 		old-locale-1.0.0.7_2 \
-		old-time-1.1.0.3_2 \
-		parsec-3.1.13.0_2 \
+		parsec-3.1.14.0_1 \
 		pem-0.2.4 \
 		prettyclass-1.0.0.0 \
-		primitive-0.6.4.0_1 \
+		primitive-0.7.0.1 \
 		random-1.1_1 \
 		raw-strings-qq-1.1 \
 		readable-0.3.1 \
-		regex-base-0.93.2_1 \
-		regex-posix-0.95.2 \
+		regex-base-0.94.0.0_1 \
+		regex-posix-0.96.0.0_1 \
 		scientific-0.3.6.2 \
-		snap-core-1.0.4.0 \
-		snap-server-1.1.1.1_1 \
-		socks-0.6.0 \
-		streaming-commons-0.2.1.1 \
-		text-1.2.3.1_2 \
-		tls-1.5.0 \
+		SHA-1.6.4.4 \
+		snap-core-1.0.4.1 \
+		snap-server-1.1.1.1_3 \
+		socks-0.6.1 \
+		streaming-commons-0.2.1.2 \
+		text-1.2.4.0 \
+		tls-1.5.4 \
 		transformers-base-0.4.5.2 \
 		transformers-compat-0.6.5 \
-		unix-compat-0.5.1_1 \
+		unix-compat-0.5.2_1 \
 		unordered-containers-0.2.10.0_1 \
 		utf8-string-1.0.1.1_3 \
-		vector-0.12.0.3_2 \
+		vector-0.12.1.2 \
 		x509-1.7.5_1 \
 		x509-store-1.6.7_1 \
 		x509-system-1.6.6 \
 		x509-validation-1.6.11 \
-		zip-archive-0.3.2.1_1 \
-		zlib-0.6.2 \
+		zip-archive-0.4.1_1 \
+		zlib-0.6.2.1_1 \
 		zlib-bindings-0.1.1.5_2
 
 post-extract:
-	${CP} ${DISTDIR}/${DIST_SUBDIR}/all-packages?${PORTNAME}-${PORTVERSION} ${WRKSRC}/elm_all-packages
+	${RM} "${WRKSRC}/worker/elm.cabal"
+	${MKDIR} "${CABAL_HOME}"
+	${TAR} -xzf "${DISTDIR}/${DIST_SUBDIR}/elm-compiler-bootstrap-${PORTVERSION}${EXTRACT_SUFX}" -C "${CABAL_HOME}"
+	${MV} "${CABAL_HOME}/elm-compiler-bootstrap-${PORTVERSION}" "${CABAL_HOME}/.elm"
+
+.PHONY: create-bootstrap
+# Create the bootstrapping elm packages and registry files
+# This is needed because it builds the reactor app, which is written in elm!
+create-bootstrap:
+	# Prepare source
+	${MKDIR} "${CABAL_HOME}"
+	${TAR} -xzf "${DISTDIR}/${DIST_SUBDIR}/${GH_ACCOUNT}-${GH_PROJECT}-${PORTVERSION}_GH${_GITHUB_REV}${EXTRACT_SUFX}" -C "${WRKDIR}"
+	${RM} "${WRKSRC}/worker/elm.cabal"
+	# Build with a custom home
+	cd "${WRKSRC}" && \
```

Comment:
I see you are replicating `extract` and `build` targets here. How about just noting in the comment that `make create-bootstrap` should only be run after successful `make`? This is how it is done for `lang/ghc`, BTW.

Reply:
Because of the way this binary is built, we agreed on IRC that it's needed in this case.